### PR TITLE
[DOCS] Update pass infra tutorial

### PR DESCRIPTION
**Pull request by zhiics**
#6144 

This PR updates the tutorial for the pass manager. It fixes some of the stale terminology used in tutorial to reflect the support for both Relay and tir. In addition, the examples for customizing optimization pipeline and debugging are moved to a separate tutorial and included in the how-to doc.

### CODE REVIEW

**Comment by zhiics** [comment]
cc @tqchen @jroesch

File: docs/dev/use_pass_infra.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            print(mod)
+
+    with relay.build_config(opt_level=3, trace=print_ir):
+            with tvm.target.create("llvm"):
+                # Perform the optimizations.
+                mod = seq(mod)
+
+
+For more pass infra related examples in Python and C++, please refer to
+`tests/python/relay/test_pass_manager.py`_ and
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:112]: Shall we put some examples in the tvm.ir.Sequential etc and refer to the API docs?

It seems to be not too useful to refer to the test code

File: docs/dev/use_pass_infra.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+.. code:: python
+
+    def print_ir(mod, info, is_before):
+        """Print the name of the pass, the IR, only before passes execute."""
+        if is_before:
+            print(f"Running pass: {}", info)
+            print(mod)
+
+    with relay.build_config(opt_level=3, trace=print_ir):
+            with tvm.target.create("llvm"):
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:106]: alignment

File: docs/dev/use_pass_infra.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+.. code:: python
+
+    def print_ir(mod, info, is_before):
+        """Print the name of the pass, the IR, only before passes execute."""
+        if is_before:
+            print(f"Running pass: {}", info)
+            print(mod)
+
+    with relay.build_config(opt_level=3, trace=print_ir):
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:105]: use tvm.transform.PassContext

File: docs/dev/use_pass_infra.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    # Users can disable any passes that they don't want to execute by providing
+    # a list, e.g. disabled_pass=["EliminateCommonSubexpr"].
+    with relay.build_config(opt_level=3):
+        with tvm.target.create("llvm"):
+            # Perform the optimizations.
+            mod = seq(mod)
+
+Debugging
+=========
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:70]: yse second level title as opposed to all the first level title.

File: docs/dev/use_pass_infra.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+passes at different layers (such as Relay and tir). This doc illustrates how we
+can use some of the more advanced features in the pass infra, e.g. customizing your
+own optimization pass pipeline and debugging. Here, we will mainly focus on
+using sequential passes. For more details about other type of passes and how
+they interact with ``Sequential``, please refer to :ref:`pass-infra`.
+
+The following example not only illustrates how users can directly create a sequential
+pass using Python APIs (this could be applied to module- and function-level
+passes as well), but also explains how we can build an optimization pipeline
+using ``Sequential`` associated with other types of passes.
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:33]: py:class:`tvm.transform.Sequantial`

File: docs/dev/use_pass_infra.rst [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+How to customize optimization pass pipeline
+===========================================
+In the :ref:`pass-infra` tutorial, we have introduced the pass infra that
+provides a systematic and consistent way for TVM to manage the optimization
+passes at different layers (such as Relay and tir). This doc illustrates how we
+can use some of the more advanced features in the pass infra, e.g. customizing your
+own optimization pass pipeline and debugging. Here, we will mainly focus on
+using sequential passes. For more details about other type of passes and how
+they interact with ``Sequential``, please refer to :ref:`pass-infra`.
```
Comment (tqchen) [inline docs/dev/use_pass_infra.rst:28]: Consider add a summary section in the end to summarize the key points

File: docs/dev/use_pass_infra.rst (hunk 1 above)
Comment (zhiics) [reply docs/dev/use_pass_infra.rst:112]: I realized that we have a tutorial already https://tvm.apache.org/docs/tutorials/dev/relay_pass_infra.html

We can probably just refactor this one and add it to how-to

**Comment by zhiics** [comment]
@tqchen I moved the debugging section to the the tutorial and updated the tutorial correspondingly. Since there are examples using different ways (including sequential) to invoke passes, I removed it from how-to and only kept the tutorial. I think that should be sufficient.

**Pull request merged**